eng/scripts/CodeChecks.ps1

```diff
@@ -94,6 +96,45 @@ try {
         & $PSScriptRoot\Export-API.ps1 -ServiceDirectory $ServiceDirectory -SDKType $SDKType -SpellCheckPublicApiSurface:$SpellCheckPublicApiSurface
     }
 
+    Write-Host "Validating installation instructions"
+    Join-Path "$PSScriptRoot/../../sdk" $ServiceDirectory  `
+        | Resolve-Path `
+        | % { Get-ChildItem $_ -Filter "README.md" -Recurse } `
+        | % {
+            $readmePath = $_
+            $readmeContent = Get-Content $readmePath
+            if ($readmeContent -Match "dotnet add .*--version.*$")
+            {
+                LogError "Specific versions should not be specified in the installation instructions. For beta versions, include the --prelease flag."
+                break
+            }
+            
+            if ($readmeContent -Match "dotnet add")
+            {
+                $changelogPath = Join-Path $(Split-Path -Parent $readmePath) "CHANGELOG.md"
+                $changeLogEntries = Get-ChangeLogEntries -ChangeLogLocation $changelogPath
+                $hasGa = $false;
+                foreach ($key in $changeLogEntries.Keys)
+                {
+                    $entry = $changeLogEntries[$key]
+                    if ($entry.ReleaseStatus -ne "(Unreleased)" -and $entry.ReleaseVersion -notmatch "beta")
+                    {
+                        $hasGa = $true;
+                        break
+                    }
+                }
+                if ($hasGa)
+                {
+                    if (-Not ($readmeContent -Match "dotnet add (?!.*--prerelease).*$")){
```

Comment:
What about the scenario where the README demonstrates the GA and beta addition?   We've been asked by service teams in the Cognitive areas to consider doing this.   We haven't yet done so or formally discussed with the architects, but it may make sense to consider it being a possibility.   For example, something like:



### Install the package

Install the Azure Text Analytics client library for .NET with [NuGet](https://www.nuget.org/):

```dotnetcli
dotnet add package Azure.AI.TextAnalytics
```

### Install the latest beta release

Features currently in preview can be explored by installing the latest beta release:

```dotnetcli
dotnet add package Azure.Messaging.EventHubs
```

Reply:
The check just ensures that there is at least a GA installation instruction if the package is GA. As the error message indicates, it is still okay to _also_ have prerelease instructions.

Reply:
my bad, then.   I thought it was enforcing that ALL instances of `dotnet add` were GA

Reply:
Nope, it is saying that if there is not at least one instance without --prerelease, fail the validation.